File: content/snippet/code-signal/arcade/intro/island-of-knowledge/minesweeper/minesweeper.py
def minesweeper(matrix):
    rows = len(matrix)
    cols = len(matrix[0])

    result = [[0 for _ in range(cols)] for _ in range(rows)]

    for row in range(rows):
        for col in range(cols):
            if matrix[row][col]:
                for i in range(max(0, row - 1), min(rows, row + 2)):
                    for j in range(max(0, col - 1), min(cols, col + 2)):
                        if i != row or j != col:
                            result[i][j] += 1
    return result


print(minesweeper([
    [True, False, False],
    [False, True, False],
    [False, False, False]
]))

File: content/snippet/code-signal/arcade/intro/island-of-knowledge/minesweeper/minesweeper.js
function minesweeper(matrix) {
  const result = [];
  for (let i = 0; i < matrix.length; i++) {
    result[i] = [];
    for (let j = 0; j < matrix[i].length; j++) {
      let count = 0;
      if (i > 0 && j > 0 && matrix[i - 1][j - 1]) count++;
      if (i > 0 && matrix[i - 1][j]) count++;
      if (i > 0 && j < matrix[i].length - 1 && matrix[i - 1][j + 1]) count++;
      if (j > 0 && matrix[i][j - 1]) count++;
      if (j < matrix[i].length - 1 && matrix[i][j + 1]) count++;
      if (i < matrix.length - 1 && j > 0 && matrix[i + 1][j - 1]) count++;
      if (i < matrix.length - 1 && matrix[i + 1][j]) count++;
      if (i < matrix.length - 1 && j < matrix[i].length - 1 && matrix[i + 1][j + 1]) count++;
      result[i][j] = count;
    }
  }
  return result;
}

console.log(
  minesweeper([
    [True, False, False],
    [False, True, False],
    [False, False, False],
  ])
);
